pkg/operator/ceph/object/user/controller.go

```diff
@@ -293,6 +293,25 @@ func (r *ReconcileObjectStoreUser) createOrUpdateCephUser(u *cephv1.CephObjectSt
 	logCreateOrUpdate := fmt.Sprintf("retrieved existing ceph object user %q", u.Name)
 	var user admin.User
 	var err error
+
+	// get the user defined k8s s3 keys if exists
+	secretName := object.GenerateCephUserSecretName(u.Spec.Store, u.Name)
+	namspacedName := types.NamespacedName{Namespace: u.Namespace, Name: secretName}
+	cephObjectStoreUserSecret := &corev1.Secret{}
+	err = r.client.Get(r.clusterInfo.Context, namspacedName, cephObjectStoreUserSecret)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			logger.Debugf("no user secret %q provided for cephobjectuser, using random values", secretName)
```

Comment:
We don't really use random values IIR. We instead get the user from RGW. If the user exists already, then Rook uses that key. If the user doesn't exist, then the user is created and IIRC gets a random key.

Reply:
Updated